Jawn/add members

jawndiego opened the pull request.

0xTranqui reviewed (changes requested):
thoughts/comments:

1. can you please update the placeholder value in the input field so it is clear you can pass in an ENS or a full address? 
2. the "search" icon currently in use for confirming a specific ens/address for inclusion would prob be suited by a different "confirmation" esque icon. idk what
3. the isInTreeStatus is not updating if I change the wallet that I have connected. I assume this is because the useCallBack isn't fire because address (from useAccount hook) isnt in the dependency array, but not too sure how useCallBack works so dont want to definitelvely say for now. i imagine that both isAdmin AND isInTree need to be rechecked whenever the authd in address changes

would you mind describing your usage of useCallback that I see youve added in other parts of this update as well? generally unfamiliar with the approach

Please revert the addReady state back to "{!!sendDataConfig ? true : false}". To be clear, this check is in here to ensure that the transaction has been properly prepared via wagmi usePrepareContractWrite. adding a dependency that is used separately from sendDataConfig (its set as "or" right now, not as "and") means we are potentially enabling invalid txns from being prepped. also, if this was to be done correctly it should have additional check for "isInTree", not just isAdmin. attached screenshot of code Im referring to below

<img width="547" alt="Screenshot 2023-09-25 at 1 16 11 PM" src="https://github.com/1ifeworld/river/assets/93691906/628bb826-ab1b-44f5-bea4-1a98d77d06ae">

jawndiego commented:
appreciate the thoughtful comments. 

1. updated
2. changed to + icon 
3. fixed! 
4. reverted the check to include ? true : false 

i was having many issues with constant re-renders and lots of checks happening with my code. switched to usecallback to prevent unnecessary re-renders of components. changed in other places of code for consistency. there may be better ways to deal with re-renders

salieflewis commented on apps/frontend/components/client/addToChannel/ChannelModal.tsx:
```diff
@@ -1,37 +1,86 @@
-'use client'
+// ChannelModal.tsx
```
Unnecessary comment.

salieflewis commented on apps/frontend/package.json:
```diff
@@ -26,11 +26,12 @@
     "lanyard": "^1.1.2",
     "lucide-react": "^0.269.0",
     "next": "13.4.19",
+    "popover": "^2.4.1",
```
Drop this dependency.

jawndiego commented on apps/frontend/utils/resolveENSorAddress.ts:
i was having problems with doing bi-directonal resolve. i.e ens to eth address and eth address to ens. it only world for eth address to ens

0xTranqui reviewed (changes requested):
things:

1. could we override the name of "client" as its used in the web3 storage hook? its confusing because there are other things like "client" we are running elsewhere (ex: viem)

ideally could be changed to something like:

const { client: **web3StorageClient** } = useWeb3Storage(uriCid)

and then of course the things this affect (wherever else client is used) would need to be updated as well

2. Im seeing that "rendererInit" was removed and zeroHash was hardcoded directly in place. we should not do this because it will make it confusing in the future when we potentially want to provide an actual "rendererInit" value. could you instead do something like "rendererInit = zeroHash", or keep the declaration of rendererInit elsewhere?


3. does the code change in the screenshot below not break the function? if we are targeting the useSetupPress hook with an object (see how there are curly braces inside the input parenthesis), wont just having factory there not work?
<img width="635" alt="Screenshot 2023-09-25 at 10 36 06 PM" src="https://github.com/1ifeworld/river/assets/93691906/36aac305-3aa2-4904-ac8f-c73ee8910daf">


4.  why are we getting rid of these errors that help explain what shappenign in the UI? is it because they arent actually explaining whats going on?
<img width="560" alt="Screenshot 2023-09-25 at 10 38 00 PM" src="https://github.com/1ifeworld/river/assets/93691906/40a343bd-aad1-47c3-9e8a-f0c0124f8cd6">

5. shouldnt the merkle tree check should only run if the user is not an admin? that way we can avoid having to do that query + boost response times. referring to apps/frontend/hooks/useIsAdminOrInTree.ts

6. confused how/why we are using the "isHex" regex function (see below). anytime we want to check if something is an address, we should use "[isAddress](https://viem.sh/docs/utilities/isAddress.html#isaddress)" in viem

<img width="158" alt="Screenshot 2023-09-25 at 10 42 17 PM" src="https://github.com/1ifeworld/river/assets/93691906/22eee15f-e339-449e-860a-258e3dea6912">

0xTranqui reviewed (approved):
approving but plz do the following:

1. delete the "isHex" function that isnt in use anymore
2. wagmi suggests you use the pattern `func?.()` for envoking write calls. this commit changes setupPress?.() to setupPress(). giving you benefit of doubt but plz explain why if ur keeping

salieflewis reviewed (approved):
LGTM.
